### ISSUE TITLE
[Feature] screenshot -raw is using gamma correction [sf#248]

### ISSUE THREAD

**Issue by openMSX-import** [openMSX/openMSX#272]
**Reported by sd-snatcher on 2012-12-19 20:32 UTC**

The "screenshot -raw" should take a raw screenshot of the MSX image, without any scaling or effects. 

But currently the image has the gamma correction effect applied over it, resulting in modified colors.

Tested on openMSX 0.9.1-dev12913 on Mac OS-X 10.6.

**Comment by openMSX-import** [comment]
**Commented by m9710797 on 2012-12-20 16:42 UTC**
This is a side effect of the rendering implementation in openMSX: color transformations (like gamma, but e.g. also the color_matrix settings) are not implemented as a post-processing step (this would be needlessly inefficient). Instead the image is directly rendered using the 'processed' colors. Or in other words, instead of rendering the image in 'standard' colors and then transform all those pixels, we transform the color palette only once and then render all pixels directly using those colors.

I agree that for an openMSX user this doesn't matter and that from his point of view gamma could be seen as a post-processing effect.

Though I don't believe it's worth changing our rendering approach to support this request. i can even imagine there are equally many users (maybe even more?) who prefer the current approach (getting a screenshot with the same colors you normally see, just without scale or blur effects).

A possible workaround is to change the color related settings to 'normal' (But what is 'normal' for the gamma setting?) Then render one frame, take the screenshot and last restore the color settings again. Though this workaround can also easily be scripted by users who care enough. Or if you can give a convincing enough use case, I'd consider making it (an optional) part of our standard screenshot script. 

So for now I'm giving this request low priority.

**Comment by openMSX-import** [comment]
**Updated by m9710797 on 2012-12-20 16:42 UTC**
- **labels**: 420819 --> 
- **priority**: 5 --> 2

**Comment by openMSX-import** [comment]
**Commented by sd-snatcher on 2012-12-22 23:32 UTC**
Ok, don't worry. I was just converting some screenshots using openMSX as a easier way. I can use another method without trouble. I thought the gamma correction was a post-processing step.

**Comment by openMSX-import** [comment]
**Updated by sd-snatcher on 2012-12-22 23:32 UTC**
- **status**: open --> closed

**Comment by openMSX-import** [comment]
**Commented by mthuurne on 2012-12-23 00:51 UTC**
You can set the gamma correction to 1.0, in that case no correction is applied.
